projects/ccf-rui/src/app/shared/components/visibility-menu/visibility-menu.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, Input, Output, EventEmitter, HostBinding } from '@angular/core';
+
+/**
+ * Interface for visibility item data
+ */
+export interface VisibilityItem {
+
+  /**
+   * Name of the item
+   */
+  name: string;
+
+  /**
+   * Whether the item is currently highlighted
+   */
+  highlighted: boolean;
+
+  /**
+   * Src of the icon (visible or non-visible)
+   */
+  iconSrc: string;
+
+  /**
+   * Opacity value
+   */
+  opacity?: number;
+}
+
+/**
+ * Menu for displaying visibility options
+ */
+@Component({
+  selector: 'ccf-visibility-menu',
+  templateUrl: './visibility-menu.component.html',
+  styleUrls: ['./visibility-menu.component.scss']
+})
+export class VisibilityMenuComponent {
+
+  /**
+   * HTML class name
+   */
+  @HostBinding('class') readonly clsName = 'ccf-visibility-menu';
+
+  /**
+   * Items to be displayed in the visibility menu
+   */
+  @Input() visibilityItems: VisibilityItem[];
+
+  /**
+   * Emits the currently highlighted items
+   */
+  @Output() highlightChange = new EventEmitter<VisibilityItem[]>();
```

Comment:
highlightChange => visibleItemsChange

Reply:
Also add a gettter `get visibleItems(): VisiblityItem[]`

Reply:
Actually this should be `@Input() visibleItems: VisibilityItem[];` so that the visibility state can be managed from the state manager.